pkgs/overlay.nix

```diff
@@ -54,15 +54,29 @@ in {
 
   docsplit = super.callPackage ./docsplit { };
 
-  elasticsearch7-oss = super.elasticsearch7-oss.overrideAttrs(_: rec {
+  elasticsearch7 = (super.elasticsearch7.override {
+    jre_headless = self.jdk11_headless;
+  }).overrideAttrs(_: rec {
+    version = elk7Version;
+    name = "elasticsearch-${version}";
+
+    src = super.fetchurl {
+      url = "https://artifacts.elastic.co/downloads/elasticsearch/${name}-linux-x86_64.tar.gz";
+      sha256 = "07p16n53fg513l4f04zq10hh5j9q6rjwz8hs8jj8y97jynvf6yiv";
+    };
+    meta.license = null;
```

Comment:
This pattern (apparently used elsewhere) needs to state why we are doing this. @dpausp noted that this is due to a bug in the unfree handling in overlays.